playbooks/ansible-tox-py39/pre.yaml

```diff
@@ -4,5 +4,5 @@
     - name: Ensure python3.9 is present
       become: true
       package:
-        name: python39-devel
+        name: '{% if (ansible_distribution == "Fedora" and ansible_distribution_major_version|int >= 34) %}python-devel{% else %}python39-devel{% endif %}'
```

Comment:
Well, this is pretty complex :-). Please write two different tasks and use a `when` condition to apply the right one depending on the OS.

Reply:
In fact there is a pattern that I documented few years back which is also used by many roles from zuul-jobs, you can read about it at https://sbarnea.com/ansible/distro-vars/

Still, in this particular case we do have a much better solutions, using a required filename instead of rpm name for installing the dependency and avoiding the change in name.